controllers/C_Document.class.php

```diff
@@ -542,8 +542,7 @@ function retrieve_action(string $patient_id = null, $document_id, $as_file = tru
 
         switch ($context) {
             case "patient_picture":
-                $this->patientService->setPid($patient_id);
-                $document_id = $this->patientService->getPatientPictureDocumentId();
+                $document_id = $this->patientService->getPatientPictureDocumentId($patient_id);
```

Comment:
I updated the PatientService to be stateless. so that it's constructor does not require a pid. Here I'm just updating dependent code.